### PR TITLE
feat(input): Add readOnly property

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #2726

## Summary

feat(input): Add readOnly property. #2726

### CODE REVIEW

File: src/components/calcite-input/calcite-input.scss [hunk 1]
```diff
@@ -24,7 +24,7 @@
   & textarea {
     @apply text--2h
     py-1
-    px-2
```
Comment (driskull) [inline src/components/calcite-input/calcite-input.scss:27]: @jcfranco I didn't run prettier or anything but some post-commit hook is doing this. It seems to have happened here: https://github.com/Esri/calcite-components/commit/f5270f8085b8a08955e65cdc9fd2a1a39c7e5e0b#diff-4185271e827d2b26e19a1a16201fb3dc4a31448b796c61107967bb5cfe2ca722R27

Maybe some are skipping these hooks???

Comment (jcfranco) [reply src/components/calcite-input/calcite-input.scss:27]: For posterity, the auto-linting NPM setup script was not using the recommended lifecycle script hook and was easy to miss. https://github.com/Esri/calcite-components/commit/566a991a96e7605bc1f8ff2c7d7f50614fbe528d updates it to set up on `npm install`.

File: src/components/calcite-input/calcite-input.e2e.ts [hunk 2]
```diff
@@ -1181,5 +1181,20 @@ describe("calcite-input", () => {
       expect(await calciteInput.getProperty("value")).toBe(initialValue);
       expect(await input.getProperty("value")).toBe(localizeNumberString(initialValue, "en-US", true));
     });
+
+    it("sets internals to readOnly or disabled when component is readOnly", async () => {
```
Comment (jcfranco) [inline src/components/calcite-input/calcite-input.e2e.ts:1185]: Could this be rewritten to test the behavior and not the implementation details?

**Review by jcfranco — APPROVED** [review]
👍🏼

**Pull request merged**